chromadb/api/models/Collection.py

```diff
@@ -150,9 +153,30 @@ def add(
         # We need to compute the embeddings if they're not provided
         if embeddings is None:
             # At this point, we know that one of documents or images are provided from the validation above
+
+            def chunk_and_assign_ids(
+                input: Embeddable, ids: IDs, chunker: Chunker[Embeddable]
+            ) -> Tuple[List[Embeddable], List[IDs]]:
+                chunks = chunker(input)
+                chunk_ids = []
+                for chunk, id in zip(chunks, ids):
+                    chunk_ids.append([f"{id}_{i}" for i in range(len(chunk))])
+                return chunks, chunk_ids
+
             if documents is not None:
+                if chunker is not None:
+                    chunks, chunk_ids = chunk_and_assign_ids(documents, ids, chunker)
+                    # Flatten chunk_ids and chunks, and assign them as the new values
+                    ids = [id for chunk_id in chunk_ids for id in chunk_id]
+                    documents = [
+                        cast(Document, doc) for chunk in chunks for doc in chunk
+                    ]
+
                 embeddings = self._embed(input=documents)
             elif images is not None:
+                if chunker is not None:
+                    # Image chunking is not supported yet
+                    raise NotImplementedError("Image chunking is not supported yet.")
                 embeddings = self._embed(input=images)
             else:
                 if uris is None:
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line 418]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->`update` should be fairly straightforward since it's a delete and an add under the hood. We only need to do the same trick with IDs, delete the old ID and insert new ones. <!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/chroma-core/chroma/2261?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->